File: java/Somatorio.java
import java.util.Arrays;

public class Somatorio {

    /**
     * Recebe um array de números e retorna a soma de todos os elementos.
     * @param arr Array de números
     * @return Soma total
     */
    public static double calcular(double[] arr) {
        if (arr == null) return 0.0;
        double total = 0.0;
        for (double num : arr) {
            total += num;
        }
        return total;
    }

    public static void main(String[] args) {
        double[] arrayExemplo = {10.0, 20.5, 30.0, -5.0, 4.5};
        System.out.println("--- Somatório ---");
        System.out.println("Array: " + Arrays.toString(arrayExemplo));
        System.out.println("Soma Total: " + calcular(arrayExemplo));
    }
}

File: javascript/somatorio.js
/**
 * Recebe um array de números e retorna a soma de todos os elementos (Sum Array).
 * @param {number[]} arr
 * @returns {number}
 */
function calcularSomatorio(arr) {
  if (!Array.isArray(arr)) {
    throw new TypeError("Esperado um array de números");
  }
  return arr.reduce((acc, curr) => acc + curr, 0);
}

// Testes de exemplo
const arrayExemplo = [10, 20.5, 30, -5, 4.5];
console.log("--- Somatório ---");
console.log(`Array: [${arrayExemplo.join(", ")}]`);
console.log(`Soma Total: ${calcularSomatorio(arrayExemplo)}`);

module.exports = { calcularSomatorio };
